### DISCUSSION TITLE
Querying QoS Profiles

### DISCUSSION THREAD

**Discussion by vidhirikin27** [camaraproject/QualityOnDemand#203] [Q&A] [answered]
Hi everyone,

I am Vidhi Mehta, Product Manager from Comviva, a Tech Mahindra company. I am new to the community and reading more about the Quality-on-Demand API. We are new to this area and hence in the exploration stage. I have a question for GET QoS Profile API. Is this API supposed to fetch the list of profiles from the exposure function or directly from the policy function? Can someone please guide me?

**Comment by eric-murray** [comment, marked as answer]
Hi Vidhi

CAMARA do not define API implementations, but do provide example implementations. For QoD implementation examples, see [here](https://github.com/camaraproject/QualityOnDemand_PI1), [here](https://github.com/camaraproject/QualityOnDemand_PI2) and [here](https://github.com/camaraproject/QualityOnDemand_PI3).

If, by "policy function", you mean PCRF / PCF, then I'm not aware of a 3GPP defined mechanism through which the PCRF / PCF can maintain a catalogue of QoS profiles that can be synchronised with the QoD API. So I would expect you would need to include that functionality in your QoD API implementation, and then map the CAMARA QoS profile parameters to their equivalents for whatever interface you are using to configure sessions using the PCRF / PCF. But talk to your PCRF / PCF vendor.

**Comment by vidhirikin27** [comment]
Thanks Eric, these references are helpful.
